app/controllers/Checkout.scala

```diff
@@ -73,22 +69,31 @@ object Checkout extends Controller with LazyLogging with ActivityTracking with C
         .fold({ case (plan, group) => Left(PlanList(plan, group.betterThan(plan).productPlans.sortBy(_.priceGBP.amount):_*)) },
               { case (plan, group) => Right(PlanList(plan, group.betterThan(plan).productPlans.sortBy(_.priceGBP.amount):_*)) })
 
+      val productPlan = planListEither.fold(identity, identity).default
       val personalData = user.map(PersonalData.fromIdUser)
-      val countryOpt = personalData.flatMap(data => data.address.country)
-      val countryGroupWithDefault = countryOpt.fold(countryGroup)(c => CountryGroup.byCountryCode(c.alpha2).getOrElse(countryGroup))
-      val country = countryOpt orElse countryGroupWithDefault.defaultCountry
-
-      val desiredCurrency = countryGroupWithDefault.currency
-      val supportedCurrencies = planListEither.fold(identity, identity).default.currencies
-      val defaultCurrency = if (supportedCurrencies.contains(desiredCurrency)) desiredCurrency else GBP
+      val productData = ProductPopulationData(user.map(_.address), planListEither)
+      val preselectedCountry = personalData.flatMap(data => data.address.country)
+      val countryGroupForPreselectedCountry = preselectedCountry.flatMap(c => CountryGroup.byCountryCode(c.alpha2))
+      val determinedCountryGroup = if (productPlan.products.seq.contains(Delivery)) {
```

Comment:
voucher won't contain delivery - you could check whether the checkout is for physical products with `planListEither.isRight`